java/src/com/googleintern/wfm/ruleengine/module/Rule.java

```diff
@@ -0,0 +1,28 @@
+package java.src.com.googleintern.wfm.ruleengine.module;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.Set;
+
+@AutoValue
+public abstract class Rule {
+    public abstract int workgroupId();
+    public abstract int casePoolId();
+    public abstract ImmutableSet<Integer> permissionSetIds();
```

Comment:
Can we include PoolAssignment here? 
public abstract PoolAssignment poolAssignment()